src/main/java/com/mael/ttt/players/HumanPlayer.java

```diff
@@ -23,24 +23,24 @@ public Mark getMark() {
         return mark;
     }
 
-    private String returnValidCellIndex() {
-        String cellIndex      = gameUI.getInput();
-        String validCellIndex = "[1-9]";
-
-        while (!cellIndex.matches(validCellIndex)) {
+    private String returnValidCellIndex(Board board) {
+        String cellIndex = gameUI.getInput();
+        while (isNotValidCellIndex(board, cellIndex)) {
             gameUI.printNotValidCellMessage();
             cellIndex = gameUI.getInput();
         }
-
         return cellIndex;
     }
 
+    private boolean isNotValidCellIndex(Board board, String cellIndex) {
+        return !gameUI.getBoardIndexesAsStrings(board.getSize()).contains(cellIndex);
+    }
+
     private String returnEmptyCellIndex(Board board, String cellIndex) {
```

Comment:
The method `isNotValidCellIndex()` was in the board at the beginning:

```java
    private String returnValidCellIndex(Board board) {
        String cellIndex = gameUI.getInput();
        while (board.isNotValidCellIndex(cellIndex)) {
            ...
        }
    }
```

but I moved it to the user interface because it converts the indexes to String in order to check the input, and why would the board have to know that I'm using a console as input and that the inputs are strings?

For example, in the method that checks that the cell is empty, I can use a method of the board, because at that point I have already checked that the input is a number (in string format, but a number), so I can safely call `Integer.parseInt()`:

```java
    private String returnEmptyCellIndex(Board board, String cellIndex) {
        while (board.isCellBusy(stringToNumber(cellIndex))) {
            ...
        }
    }
```

Do you still think I should move `isNotValidCellIndex()` back to the Board?